annif/suggestion.py

```diff
@@ -175,12 +181,11 @@ def vector(self):
 
     def filter(self, limit=None, threshold=0.0):
         hits = sorted(self.hits, key=lambda hit: hit.score, reverse=True)
+        hits = [hit for hit in hits if
```

Comment:
The `hits` variable gets reassigned here. I think it would be better to use a different variable name. The first one could be e.g. `sorted_hits`.